lib/src/test/kotlin/org/didcommx/didcomm/crypto/key/RecipientKeySelectorTest.kt

```diff
@@ -217,4 +234,168 @@ class RecipientKeySelectorTest {
             assertEquals("The recipient '$bobDIDUrl' curve is not compatible to 'X25519'", actual.message)
         }
     }
+
+    @Test
+    fun `Test_find_anoncrypt_pack_recipient_public_keys_by_did_positive`() {
+        val keySelector = RecipientKeySelector(DIDDocResolverMock(), getSecretsResolver(Person.BOB))
+
+        val expected = getKeyAgreementMethods(Person.BOB, KeyAgreementCurveType.X25519)
+        val res = keySelector.findAnonCryptKeys(listOf(JWM.BOB_DID)).toList()
+
+        val exp = expected.map { vm -> Key.wrapVerificationMethod(vm) }
+
+        assertEquals(res, exp)
+        assertEquals(expected[0], DID_DOC_BOB_WITH_NO_SECRETS.verificationMethods[0])
+    }
+
+    @Test
+    fun `Test_find_anoncrypt_pack_recipient_public_keys_by_kid_positive`() {
+        val keySelector = RecipientKeySelector(DIDDocResolverMock(), getSecretsResolver(Person.BOB))
+
+        DID_DOC_BOB_WITH_NO_SECRETS.verificationMethods.forEach { vm ->
+            val res = keySelector.findAnonCryptKeys(listOf(vm.id)).map { it.jwk.toPublicJWK() }
+//            res.forEach { key -> key.jwk = key.jwk.toPublicJWK() }
+            val listOfVM = listOf(Key.wrapVerificationMethod(vm))
+            assertTrue { ReflectionEquals(res).matches(listOfVM) }
```

Comment:
If you need to compare arrays, probably you can use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/content-equals.html or https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/content-deep-equals.html ?